services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/service/implement/SubnetServiceImp.java

```diff
@@ -313,4 +318,31 @@ public void addSubnetIdToVpc(String subnetId, String projectId, String vpcId) th
 
     }
 
+    @Override
+    public boolean checkIfCidrOverlap(String cidr,String projectId, String vpcId) throws FallbackException, ResourceNotFoundException, ResourcePersistenceException, CidrNotWithinNetworkCidr {
+
+        // get vpc and check with vpc cidr
+        VpcWebJson vpcWebJson = verifyVpcId(projectId, vpcId);
+        String vpcCidr = vpcWebJson.getNetwork().getCidr();
+
+        if (!SubnetManagementUtil.IsCidrWithin(cidr, vpcCidr)) {
```

Comment:
vpcCidr could be empty or null for OpenStack network.

Reply:
Also we have different handling of subnet-vpc overlap and subnet-subnet overlap. May I know why?